gtsfm/densify/interface_patchmatchnet.py

```diff
@@ -0,0 +1,195 @@
+"""Interface class from GtsfmData to PatchmatchNetData
+
+Authors: Ren Liu
+"""
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import numpy as np
+from torch.utils.data import Dataset
+
+from gtsfm.common.image import Image
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.densify.mvs_math import piecewise_gaussian
+
+NUM_PATCHMATCHNET_STAGES = 4
+
+
+class PatchmatchNetData(Dataset):
+    """PatchmatchNetData class for PatchmatchNet. It contains the interface from GtsfmData.
+    Wang's work in https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/datasets/dtu_yao_eval.py is referred to.
+    """
+
+    def __init__(self, images: Dict[int, Image], sfm_result: GtsfmData, num_views: int = 5) -> None:
+        """Initialize method for PatchmatchNetData
+
+        Args:
+            images: input images (H, W, C) to GTSFM
+            sfm_result: sfm results calculated by GTSFM
+            num_views: number of views, containing 1 reference view and (num_views-1) source views
+        """
+        assert images is not None and len(images) > 1
+
+        # cache sfm result
+        self._sfm_result = sfm_result
+
+        # PatchmatchNet meta
+        self._num_views = num_views
+        self._num_stages = NUM_PATCHMATCHNET_STAGES
+
+        # Test data preparation
+        self._keys = sorted(self._sfm_result.get_valid_camera_indices())
+        self._num_images = len(self._keys)
+        self._keys_map = {}
+        self._camera_centers = {}
+        for i in range(self._num_images):
+            self._keys_map[self._keys[i]] = i
+            extrinsic = self._sfm_result.get_camera(self._keys[i]).pose().inverse().matrix()
+            rotation = extrinsic[:3, :3]
+            translation = extrinsic[:3, 3:4]
+            self._camera_centers[self._keys[i]] = (-rotation.T @ translation).flatten()
```

Comment:
@codyly, it looks like you dont need the extrinsic here. you can just read the camera center's location in the world frame directly from the pose:
```python
self._camera_centers[self._keys[i]] = self._sfm_result.get_camera(self._keys[i]).pose().translation()
```
Currently you are saving the camera center as the inverse of the inverse, and it's unnecessary.